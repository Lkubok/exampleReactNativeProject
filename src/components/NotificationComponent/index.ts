import Component from './NotificationComponent';

export default Component;
