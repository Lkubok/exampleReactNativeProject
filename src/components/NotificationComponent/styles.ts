import { StyleSheet } from 'react-native';

import { variables, fontNames } from 'res';

export default StyleSheet.create({
  body: {
    color: variables.colors.black,
    fontFamily: fontNames.SFCompactTextRegular,
    fontSize: variables.fontSize.base,
  },
  container: {
    flexDirection: 'row',
    height: 100,
    width: '100%',
  },
  iconContainer: {
    justifyContent: 'center',
    marginHorizontal: 12,
  },
  notificationIcon: {
    borderRadius: variables.borderRadius.base,
    height: 40,
    width: 40,
  },
  textWrapper: {
    justifyContent: 'center',
  },
  title: {
    color: variables.colors.black,
    fontFamily: fontNames.SFCompactTextBold,
    fontSize: variables.fontSize.medium,
    marginBottom: variables.spacing.base,
  },
});
