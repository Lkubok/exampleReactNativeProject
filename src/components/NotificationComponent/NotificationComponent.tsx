/* eslint-disable @typescript-eslint/no-unsafe-assignment */
/* eslint-disable @typescript-eslint/no-var-requires */
import React, { FC } from 'react';
import { View, Text, TouchableOpacity, Image } from 'react-native';

import styles from './styles';

const notificationIconLogo = require('../../assets/notificationIcon.png');

interface Props {
  onPress: () => void;
  title: string;
  body: string;
}

const NotificationComponent: FC<Props> = ({ onPress, title, body }) => {
  return (
    <TouchableOpacity style={styles.container} onPress={onPress} hitSlop={{ bottom: 12, left: 12, right: 12, top: 12 }}>
      <View style={styles.iconContainer}>
        <Image resizeMode="contain" source={notificationIconLogo} style={styles.notificationIcon} />
      </View>
      <View style={styles.textWrapper}>
        <Text style={styles.title}>{title}</Text>
        <Text style={styles.body}>{body}</Text>
      </View>
    </TouchableOpacity>
  );
};

export default NotificationComponent;
