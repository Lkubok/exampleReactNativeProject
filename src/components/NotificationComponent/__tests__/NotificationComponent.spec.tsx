import React from 'react';
import { act, customRender } from 'utils/testUtils';

import NotificationComponent from '../NotificationComponent';

jest.useFakeTimers();

describe('NotificationComponent component', () => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let props: any;
  beforeEach(() => {
    props = {
      disabled: true,
      loading: false,
      onPress: () => {},
      title: 'testingNotificationComponent',
      useNativeDriver: true,
    };
  });

  it('should match snapshot', () => {
    const snapshot = customRender(<NotificationComponent {...props} />).toJSON();
    // eslint-disable-next-line no-void
    void act(() => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
