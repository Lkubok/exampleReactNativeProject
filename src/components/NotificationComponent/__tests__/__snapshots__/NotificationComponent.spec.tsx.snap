// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationComponent component should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 12,
        "left": 12,
        "right": 12,
        "top": 12,
      }
    }
    onClick={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "height": 100,
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "justifyContent": "center",
          "marginHorizontal": 12,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../src/assets/notificationIcon.png",
          }
        }
        style={
          Object {
            "borderRadius": 30,
            "height": 40,
            "width": 40,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "[7m[38;2;0;0;0m#000000[39m[27m",
            "fontFamily": "SFCompactText-Bold",
            "fontSize": 18,
            "marginBottom": 12,
          }
        }
      >
        testingNotificationComponent
      </Text>
      <Text
        style={
          Object {
            "color": "[7m[38;2;0;0;0m#000000[39m[27m",
            "fontFamily": "SFCompactText-Regular",
            "fontSize": 17,
          }
        }
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;
